File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)

    local length = #a
    if length ~= #b then
        return -1
    end

    local distance = 0
    for i = 1, #a do
        if a:sub(i, i) ~= b:sub(i, i) then
            distance = distance + 1
        end
    end

    return distance
end

return Hamming

File: solutions/go/hamming/1/hamming.go
package hamming

import "errors"

// Distance calculates the differences between two strings
//
// Only strings with the same length are supported
func Distance(a, b string) (int, error) {

	if len(a) != len(b) {
		return 0, errors.New("strings must be of same length")
	}

	dist := 0
	for i, s := range a {
		if s != rune(b[i]) {
			dist++
		}
	}

	return dist, nil

}
